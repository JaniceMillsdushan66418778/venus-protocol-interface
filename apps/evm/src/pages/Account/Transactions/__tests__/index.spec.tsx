import { waitFor } from '@testing-library/dom';
import fakeAccountAddress from '__mocks__/models/address';
import { useGetAccountTransactionHistory } from 'clients/api';
import { type UseIsFeatureEnabledInput, useIsFeatureEnabled } from 'hooks/useIsFeatureEnabled';
import { en } from 'libs/translations';
import { renderComponent } from 'testUtils/render';
import type { Mock } from 'vitest';
import { Transactions } from '..';

describe('Transactions', () => {
  beforeEach(() => {
    (useIsFeatureEnabled as Mock).mockImplementation(
      ({ name }: UseIsFeatureEnabledInput) => name === 'transactionHistory',
    );
  });
  it('displays content correctly', async () => {
    const { container, getByText } = renderComponent(<Transactions />, {
      accountAddress: fakeAccountAddress,
    });
    await waitFor(() => expect(getByText(en.account.transactions.txType.mint)));

    expect(container.textContent).toMatchSnapshot();
  });

  it('displays placeholder when there are no transactions to display', async () => {
    (useGetAccountTransactionHistory as Mock).mockImplementation(() => ({
      data: {
        count: 0,
        transactions: [],
      },
      isLoading: false,
    }));
    const { container } = renderComponent(<Transactions />, {
      accountAddress: fakeAccountAddress,
    });

    expect(container.textContent).toMatchSnapshot();
  });
});
