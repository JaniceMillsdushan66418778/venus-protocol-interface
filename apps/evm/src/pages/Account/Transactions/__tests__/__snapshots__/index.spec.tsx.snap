// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Transactions > displays content correctly 1`] = `"All transactionsAll marketsAug 23, 2024Withdraw04:17 AM> 100T USDT$24.06K • vUSDT • MetaverseViewAug 06, 2024Supply02:18 AM> 100T USDT$24.01K • vUSDT • MetaverseViewAug 05, 2024Borrow02:18 AM> 100T USDT$24.01K • vUSDT • MetaverseViewAug 04, 2024Repayment02:18 AM> 100T USDT$24.01K • vUSDT • MetaverseViewAug 03, 2024Enabled collateral02:18 AMEnabled vUSDT as collateralMetaverseViewDisabled collateral02:18 AMDisabled vUSDT as collateralMetaverseViewAug 02, 2024Approval02:18 AMApproved spending > 100T vUSDTvUSDT • MetaverseView"`;

exports[`Transactions > displays placeholder when there are no transactions to display 1`] = `"All transactionsAll marketsNo transactions yetYour transactions will appear here"`;
